src/main/java/com/base/BaseScheduledFlow.java

```diff
@@ -0,0 +1,187 @@
+import com.getbase.services.StagesService;
+import com.getbase.sync.Meta;
+import com.getbase.sync.Sync;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.core.env.MissingRequiredPropertiesException;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Created by przemek on 19.10.2016.
+ */
+
+@Service
+@Slf4j
+public class BaseScheduledFlow {
+
+    private Client client;
+    private String deviceUUID;
+
+    public BaseScheduledFlow() {
+        log.info("BaseScheduledFlow service launching...");
+        deviceUUID = getProperty("deviceUUID");
+        this.client = new Client(new Configuration.Builder()
+                .accessToken(getProperty("BASECRM_ACCESS_TOKEN"))
+                .build());
+    }
+
+    @Scheduled(fixedDelay = 15000)
+    public void reportCurrentTime() {
+        Sync sync = new Sync(client, deviceUUID);
+        sync.subscribe(Account.class, (meta, account) -> true)
+                .subscribe(Address.class, (meta, address) -> true)
+                .subscribe(AssociatedContact.class, (meta, associatedContact) -> true)
+                .subscribe(Contact.class, this::processContact)
+                .subscribe(Deal.class, this::processDeal)
+                .subscribe(LossReason.class, (meta, lossReason) -> true)
+                .subscribe(Note.class, (meta, note) -> true)
+                .subscribe(Pipeline.class, (meta, pipeline) -> true)
+                .subscribe(Source.class, (meta, source) -> true)
+                .subscribe(Stage.class, (meta, stage) -> true)
+                .subscribe(Tag.class, (meta, tag) -> true)
+                .subscribe(Task.class, (meta, task) -> true)
+                .subscribe(User.class, (meta, user) -> true)
+                .subscribe(Lead.class, (meta, lead) -> true)
+                .fetch();
+    }
+
+    private boolean processContact(final Meta meta, final Contact contact) {
+        MDC.put("contact", contact.getId().toString());
+        final String syncEventType = meta.getSync().getEventType();
+        boolean isContactCreated = "created".equals(syncEventType);
+
+        if (isContactCreated) {
+            log.info("processing contact, sync event received: '{}'", syncEventType);
+            if ("created".equals(syncEventType)) {
+                log.info("contact with id: {} is to be verified", contact.getId());
+                if (verifyContact(contact)) {
+                    createDeal(contact);
+                }
+            }
+        } else {
+            log.info("processing contact, received sync event != 'created'. Nothing to do");
+        }
+
+        clearSync(meta);
+        MDC.clear();
+        return true;
+    }
+
+    private void clearSync(Meta meta) {
```

Comment:
Is this method necessary?

AFAIK, if using Java API acknowledgments are implicitly sent anyway.